From tklauser@distanz.ch  Wed May  5 16:03:58 2010
From: Tobias Klauser <tklauser@distanz.ch>
Date: Wed,  5 May 2010 11:18:41 +0200
Subject: USB: isp1760: Use resource_size
To: gregkh@suse.de
Cc: linux-usb@vger.kernel.org, kernel-janitors@vger.kernel.org, Tobias Klauser <tklauser@distanz.ch>
Message-ID: <1273051121-12861-1-git-send-email-tklauser@distanz.ch>


Use the resource_size function instead of manually calculating the
resource size.  This reduces the chance of introducing off-by-one
errors.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/isp1760-if.c |   11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

--- a/drivers/usb/host/isp1760-if.c
+++ b/drivers/usb/host/isp1760-if.c
@@ -36,7 +36,7 @@ static int of_isp1760_probe(struct of_de
 	struct resource memory;
 	struct of_irq oirq;
 	int virq;
-	u64 res_len;
+	resource_size_t res_len;
 	int ret;
 	const unsigned int *prop;
 	unsigned int devflags = 0;
@@ -45,13 +45,12 @@ static int of_isp1760_probe(struct of_de
 	if (ret)
 		return -ENXIO;
 
-	res = request_mem_region(memory.start, memory.end - memory.start + 1,
-			dev_name(&dev->dev));
+	res_len = resource_size(&memory);
+
+	res = request_mem_region(memory.start, res_len, dev_name(&dev->dev));
 	if (!res)
 		return -EBUSY;
 
-	res_len = memory.end - memory.start + 1;
-
 	if (of_irq_map_one(dp, 0, &oirq)) {
 		ret = -ENODEV;
 		goto release_reg;
@@ -92,7 +91,7 @@ static int of_isp1760_probe(struct of_de
 	return ret;
 
 release_reg:
-	release_mem_region(memory.start, memory.end - memory.start + 1);
+	release_mem_region(memory.start, res_len);
 	return ret;
 }
 
